### PR TITLE
[autogluon][build] AutoGluon 0.7.0 inference containers v2 - change to torch-serve

### PR DESCRIPTION
**Pull request by gradientsky**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [x] I have run builds/tests on my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by arjkesh — COMMENTED** [review]
It looks like there are some pip check failures and security failures in sanity tests - would you mind looking into those and resolving?

File: autogluon/inference/docker/0.7/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -10,12 +10,10 @@ LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
...
 
 LABEL maintainer="Amazon AI"
-LABEL dlc_major_version="1"
+LABEL dlc_major_version="2"
 
 # PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK - see https://github.com/autogluon/autogluon/issues/2534
-ENV MXNET_CUDNN_LIB_CHECKING=0 \
-    MXNET_CUDNN_AUTOTUNE_DEFAULT=0 \
-    PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK=1
+ENV PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK=1
```
Comment (tonyhoo) [inline autogluon/inference/docker/0.7/py3/Dockerfile.cpu:16]: Do we still need this with torch 11.3 and CUDA 11.7 can you verify?

File: autogluon/inference/docker/0.7/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -54,14 +31,27 @@ RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.p
...
 RUN rm -rf /usr/local/lib/python3.9/dist-packages/gluonts/nursery/tsbench
 
 # Remove these to address security issues; these are not used if only python package is used
+# flask Flask-RESTful nvgpu are reqiurements for torch-serve
 RUN rm -rf /usr/local/lib/python3.9/dist-packages/ray/jars \
- && rm -rf /opt/conda/lib/python3.9/site-packages/ray/jars
+ && rm -rf /opt/conda/lib/python3.9/site-packages/ray/jars \
+ && pip install -U flask Flask-RESTful nvgpu \
+ && conda update -y requests \
+ && pip install -U botocore pyopenssl google-auth awscli \
```
Comment (tonyhoo) [inline autogluon/inference/docker/0.7/py3/Dockerfile.cpu:45]: are these needed for any specific reason?

Comment (gradientsky) [reply autogluon/inference/docker/0.7/py3/Dockerfile.cpu:45]: those were flagged as security vulnerabilities (dependencies are coming from torch-serve)

File: autogluon/inference/docker/0.7/py3/Dockerfile.cpu [hunk 3]
```diff
@@ -54,14 +31,27 @@ RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.p
...
 
 # Removing GluonTS nursery/tsbench package - it is not used in training/inference and have security vulnerabilities
 RUN rm -rf /usr/local/lib/python3.9/dist-packages/gluonts/nursery/tsbench
 
 # Remove these to address security issues; these are not used if only python package is used
+# flask Flask-RESTful nvgpu are reqiurements for torch-serve
 RUN rm -rf /usr/local/lib/python3.9/dist-packages/ray/jars \
- && rm -rf /opt/conda/lib/python3.9/site-packages/ray/jars
+ && rm -rf /opt/conda/lib/python3.9/site-packages/ray/jars \
+ && pip install -U flask Flask-RESTful nvgpu \
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply autogluon/inference/docker/0.7/py3/Dockerfile.cpu:40]: it's already there and required by torch-serve metrics collector. I tried removing it, the container won't start.

File: autogluon/inference/docker/0.7/py3/Dockerfile.cpu (hunk 1 above)
Comment (gradientsky) [reply autogluon/inference/docker/0.7/py3/Dockerfile.cpu]: just double-checked looks like it's working now; removed; closed [#2534](https://github.com/autogluon/autogluon/issues/2534)

File: autogluon/inference/docker/0.7/py3/cu117/Dockerfile.gpu [hunk 4]
```diff
@@ -13,12 +13,10 @@ LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
...
 
 LABEL maintainer="Amazon AI"
-LABEL dlc_major_version="1"
+LABEL dlc_major_version="2"
 
 # PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK - see https://github.com/autogluon/autogluon/issues/2534
-ENV MXNET_CUDNN_LIB_CHECKING=0 \
-    MXNET_CUDNN_AUTOTUNE_DEFAULT=0 \
-    PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK=1
+ENV PYTORCH_SKIP_CUDNN_COMPATIBILITY_CHECK=1
```
Comment (gradientsky) [inline autogluon/inference/docker/0.7/py3/cu117/Dockerfile.gpu]: removed

File: autogluon/build_artifacts/inference/config.properties [hunk 5]
```diff
@@ -2,4 +2,25 @@ vmargs=-XX:+UseContainerSupport -XX:InitialRAMPercentage=8.0 -XX:MaxRAMPercentag
 model_store=/opt/ml/model
 load_models=ALL
 inference_address=http://0.0.0.0:8080
-management_address=http://0.0.0.0:8081
\ No newline at end of file
+management_address=http://0.0.0.0:8081
+enable_envvars_config=true
+# management_address=unix:/tmp/management.sock
```
Comment (yinweisu) [inline autogluon/build_artifacts/inference/config.properties:7]: Do we want to keep these?

File: autogluon/inference/docker/0.7/py3/Dockerfile.cpu (hunk 1 above)
Comment (gradientsky) [reply autogluon/inference/docker/0.7/py3/Dockerfile.cpu]: removed; confirmed 1.13.1 resolved [#2534](https://github.com/autogluon/autogluon/issues/2534)

File: autogluon/inference/docker/0.7/py3/Dockerfile.cpu (hunk 3 above)
[the remark this answers is not in the document]
Comment (gradientsky) [reply autogluon/inference/docker/0.7/py3/Dockerfile.cpu:40]: yes, torch-serve requires these for metrics reporting. Container fails if it's not there.

File: autogluon/build_artifacts/inference/config.properties (hunk 5 above)
Comment (gradientsky) [reply autogluon/build_artifacts/inference/config.properties:7]: yes; for documentation purposes (it's the same config as in pytorch images)

**Comment by gradientsky** [comment]
> It looks like there are some pip check failures and security failures in sanity tests - would you mind looking into those and resolving?

Please ignore failing `dlc-pr-autogluon-training_dlc-pr-sanity-test` - we are releasing only inference images

**Review by tonyhoo — DISMISSED** [review]

File: autogluon/inference/docker/0.7/py3/Dockerfile.cpu (hunk 3 above)
[the remark this answers is not in the document]
Comment (arjkesh) [reply autogluon/inference/docker/0.7/py3/Dockerfile.cpu:40]: Gotcha, what I meant is that it should already be there in the base image, as we install requirements directly from TS based on TS version https://github.com/aws/deep-learning-containers/blob/master/pytorch/inference/docker/1.13/py3/cu117/Dockerfile.gpu#L223

The comment is misleading, but nvgpu should be in the base GPU DLC already

For CPU, do we also need nvgpu?

[the remark this answers is not in the document]
Comment (gradientsky) [reply autogluon/inference/docker/0.7/py3/Dockerfile.cpu]: nvgpu is in the CPU container is coming from the base image. I removed it, looks like it's not needed (verified running notebooks on locally-built images).

**Comment by gradientsky** [comment]
Tests are passing; rebased onto main and reverted toml back to main version.

**Review by arjkesh — DISMISSED** [review]

**Review by tonyhoo — DISMISSED** [review]

**Review by arjkesh — APPROVED** [review]

**Pull request merged**